### PR TITLE
Allow Flow entity state to be recreated

### PR DESCRIPTION
**Pull request by blizzz**
An interface is added that allows entities to ex- and import their state identifiers, so usages in future requests is possible. The use case is notification handling: needed to solve https://github.com/nextcloud/flow_notifications/issues/1

### CODE REVIEW

File: apps/workflowengine/lib/Entity/File.php [hunk 1]
```diff
@@ -220,4 +234,40 @@ public function getUrl(): string {
 			return '';
 		}
 	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function exportContextIDs(): array {
+		$nodeOwner = $this->getNode()->getOwner();
+		//$actingUserId = null;
```
Comment (MorrisJobke) [inline apps/workflowengine/lib/Entity/File.php:243]:
```suggestion
		$actingUserId = null;
```

File: apps/workflowengine/lib/Entity/File.php [hunk 2]
```diff
@@ -220,4 +234,40 @@ public function getUrl(): string {
...
+		if ($this->actingUser instanceof IUser) {
+			$actingUserId = $this->actingUser->getUID();
+		} elseif ($this->userSession->getUser() instanceof IUser) {
+			$actingUserId = $this->userSession->getUser()->getUID();
+		}
+		return [
+			'eventName' => $this->eventName,
+			'nodeId' => $this->getNode()->getId(),
+			'nodeOwnerId' => $nodeOwner ? $nodeOwner->getUID() : null,
+			'actingUserId' => $actingUserId ?? null,
```
Comment (MorrisJobke) [inline apps/workflowengine/lib/Entity/File.php:253]:
```suggestion
			'actingUserId' => $actingUserId,
```

**Review by MorrisJobke — APPROVED** [review]
Beside the comments it looks good 👍

**Review by skjnldsv — APPROVED** [review]
:elephant:

**Pull request merged**